packages/core/src/external/carequality/ihe-gateway-v2/outbound/xcpd/process/xcpd-response.ts

```diff
@@ -49,7 +49,12 @@ function convertIheIdentifierToPersonalIdentifier(identifier: IheIdentifier): Pe
   };
 }
 
-function iheIdentifiersToPersonalIdentifiers(otherIds: IheIdentifier[]): PersonalIdentifier[] {
+function iheIdentifiersToPersonalIdentifiers(
+  otherIds: IheIdentifier[]
+): PersonalIdentifier[] | undefined {
+  if (!otherIds || otherIds.every(id => !id._extension && !id._root)) {
+    return undefined;
```

Comment:
added checks for if the entire identifier is undefined to return undefined. ditto for telecom and addreses

Reply:
Not only that, but also `convertIheIdentifierToPersonalIdentifier` could be called from somewhere else and it could result in this `PersonalIdentifier`, which doesn't look valid:

```ts
{
  value: undefined,
  system: undefined
}
```